app/models/playlist.rb

```diff
@@ -0,0 +1,10 @@
+class Playlist < ApplicationRecord
+  validates :name, presence: true
+  validate :has_content
+  def has_content
+    if genre.blank? && energy.blank? && danceability.blank?
+      errors[:base] = 'Must have a filename or a URL'
+    end
+  end
+  validates :danceability, numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: 100}
```

Comment:
Pourquoi l'avoir fait sur `danceability` mais pas `energy` ?

Reply:
De plus, ici on ne veut que des entiers, et la valeur peut-être nulle

```suggestion
  validates :danceability, numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: 100, only_integer: true, allow_nil: true }
```